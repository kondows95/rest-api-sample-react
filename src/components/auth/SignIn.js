import React from 'react'
import PropTypes from 'prop-types'
import { Box, Container, Link, Button, Grid, TextField}  from '@material-ui/core'
import FormTitle from './FormTitle';
import { makeStyles } from '@material-ui/core/styles';
import { green } from '@material-ui/core/colors';
import CircularProgress from '@material-ui/core/CircularProgress';
import { FormattedMessage } from 'react-intl';

const useStyles = makeStyles(theme => ({
  btnWrapper: {
    margin: theme.spacing(1),
    position: 'relative',
  },
  btnProgress: {
    color: green[500],
    position: 'absolute',
    top: '50%',
    left: '50%',
    marginTop: -12,
    marginLeft: -12,
  },
}))

const SignIn = ({
  authState, 
  changeAuthState,
  signIn,
  loading,
  error
}) => {
  const classes = useStyles();
  
  const [form, setForm] = React.useState({email:"", password:""})
  
  const handleChangeValue = fieldName => event => {
    const newForm = {...form}
    newForm[fieldName] =  event.target.value
    setForm(newForm)
  }
  
  const handleSubmit = event => {
    event.preventDefault()
    signIn(form['email'], form['password'])
  }
  
  const handleForgotPassword = event => {
    event.preventDefault()
    changeAuthState('forgotPassword')
  }
  
  const handleSignUp = event => {
    event.preventDefault()
    changeAuthState('signUp')
  }
  
  const content = (
    <Container component="main" maxWidth="xs">
      <form onSubmit={handleSubmit}>
        <Box display="flex" flexDirection="column" mt={8}>
          <FormTitle><FormattedMessage id="Label.SignIn" defualtMessage="Sign In" /></FormTitle>
          <Box display="flex" justifyContent="center" fontWeight={600} color="error.main">
            {error}
          </Box>
          <Box width="100%" my={2}>
            <TextField
              id="email"
              autoComplete="email"
              type="email"
              onChange={handleChangeValue('email')}
              value={form.email}
              label={loading? " " : "Email Address"}
              variant="outlined"
              required
              fullWidth
            />
          </Box>
          <Box width="100%" my={2}>
            <TextField
              id="password"
              label={loading? "" : "Password"}
              type="password"
              autoComplete="current-password"
              onChange={handleChangeValue('password')} 
              value={form.password}
              variant="outlined"
              required
              fullWidth
            />
          </Box>
          <Box width="100%" mt={4} mb={2}>
            
            <Box className={classes.btnWrapper} >
              <Button type="submit" fullWidth disabled={loading} color="primary" variant="contained">
                <FormattedMessage id="Button.SignIn" defualtMessage="Sign In" />
              </Button>
              {loading && <CircularProgress size={24} className={classes.btnProgress} />}
            </Box>
            
          </Box>
          <Box width="100%" my={2}>
            <Grid container>
              <Grid item xs>
                <Link href="#" variant="body2" onClick={handleForgotPassword}>
                  Forgot password?
                </Link>
              </Grid>
              <Grid item>
                <Link href="#" variant="body2" onClick={handleSignUp}>
                  No account? Sign Up
                </Link>
              </Grid>
            </Grid>
          </Box>
        </Box>
      </form>
    </Container>
  )
  return (authState === 'signIn') ? content : null
}

SignIn.propTypes = {
  authState: PropTypes.string.isRequired,
  changeAuthState: PropTypes.func.isRequired,
  signIn: PropTypes.func.isRequired
}

export default SignIn
