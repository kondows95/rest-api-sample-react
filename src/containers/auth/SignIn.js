import { connect } from 'react-redux'
import SignIn from '../../components/auth/SignIn'
import { changeAuthState, signIn } from '../../modules/auth.ts'

export default connect(
  (state) => ({
    authState: state.auth.authState,
    loading: state.auth.loading,
    error: state.auth.error,
  }),
  (dispatch) => ({
    changeAuthState: (value) => dispatch(changeAuthState(value)),
    signIn: (email, password) => dispatch(signIn(email, password))
  })
)(SignIn)
