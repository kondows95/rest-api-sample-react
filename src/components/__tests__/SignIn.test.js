import React from 'react'
import ReactDOM from 'react-dom';
import { act } from 'react-dom/test-utils';
import renderer from 'react-test-renderer';
import SignIn from '../auth/SignIn';

let container
beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  document.body.outerHTML = '<body></body>'
  container = null
})

const valueSetter = Object.getOwnPropertyDescriptor(window.HTMLInputElement.prototype, "value").set;
let authState = "signIn";
const changeAuthState = jest.fn((data) => {
  authState = data;
});
const loading = false;
const error = "";
const signIn = jest.fn((email1, password1) => {
  let email = email1;
  let password = password1;
});
describe("SignIn component snapshot", () => {
  it('matches the snapshot', () => {
    const SignInSnapshot = renderer.create(
      <SignIn
        authState={authState}
        changeAuthState={changeAuthState}
        loading={loading}
        error={error}
        signIn={signIn}
      />
    ).toJSON();
    expect(SignInSnapshot).toMatchSnapshot();
  });
});

describe("Testing SignIn", () => {
  it('Testing textfied and  SignIn button ', () => {
    act(() => {
      ReactDOM.render(
        <SignIn
          authState={authState}
          changeAuthState={changeAuthState}
          loading={loading}
          error={error}
          signIn={signIn}
        />, container);
    });
    const inputArr = document.querySelectorAll('input');
    act(() => {
      valueSetter.call(inputArr[0], 'a@gmail.com');
      inputArr[0].dispatchEvent(new Event('change', { bubbles: true }));
    });
    expect(inputArr[0].value).toBe('a@gmail.com');
    act(() => {
      valueSetter.call(inputArr[1], '1234');
      inputArr[1].dispatchEvent(new Event('change', { bubbles: true }));
    });
    expect(inputArr[1].value).toBe('1234');
    const buttonArr = document.querySelectorAll('button');
    act(() => {
      buttonArr[0].dispatchEvent(new MouseEvent('click', { bubbles: true }));
    });
    expect(signIn).toHaveBeenCalledWith("a@gmail.com", "1234");
  })
  it("Testing two link ", () => {
    act(() => {
      ReactDOM.render(
        <SignIn
          authState={authState}
          changeAuthState={changeAuthState}
          loading={loading}
          error={error}
          signIn={signIn}
        />, container);
    });
    const linkArr = document.querySelectorAll('a');
    act(() => {
      linkArr[0].dispatchEvent(new MouseEvent('click', { bubbles: true }));
    });
    expect(changeAuthState).toHaveBeenCalledWith("forgotPassword");
    act(() => {
      linkArr[1].dispatchEvent(new MouseEvent('click', { bubbles: true }));
    });
    expect(changeAuthState).toHaveBeenCalledWith("signUp");
  });
});
