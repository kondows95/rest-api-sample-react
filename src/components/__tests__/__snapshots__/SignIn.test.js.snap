// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn component snapshot matches the snapshot 1`] = `
<main
  className="MuiContainer-root MuiContainer-maxWidthXs"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="MuiBox-root MuiBox-root-11"
    >
      <div
        className="MuiBox-root MuiBox-root-12"
      >
        <div
          className="MuiBox-root MuiBox-root-13"
        >
          <div
            className="MuiBox-root MuiBox-root-14"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <g
                fill="none"
              >
                <path
                  d="M0 0h24v24H0V0z"
                />
                <path
                  d="M0 0h24v24H0V0z"
                  opacity=".87"
                />
              </g>
              <path
                d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
              />
            </svg>
          </div>
        </div>
        <div
          className="MuiBox-root MuiBox-root-24"
        >
          <span>
            Sign In
          </span>
        </div>
      </div>
      <div
        className="MuiBox-root MuiBox-root-25"
      >
        
      </div>
      <div
        className="MuiBox-root MuiBox-root-26"
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
            data-shrink={false}
            htmlFor="email"
          >
            Email Address
            <span
              className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            onClick={[Function]}
          >
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-83 MuiOutlinedInput-notchedOutline"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="PrivateNotchedOutline-legend-84"
                style={
                  Object {
                    "width": 0.01,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
            <input
              aria-invalid={false}
              autoComplete="email"
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              id="email"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="email"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        className="MuiBox-root MuiBox-root-85"
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
            data-shrink={false}
            htmlFor="password"
          >
            Password
            <span
              className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            onClick={[Function]}
          >
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-83 MuiOutlinedInput-notchedOutline"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="PrivateNotchedOutline-legend-84"
                style={
                  Object {
                    "width": 0.01,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
            <input
              aria-invalid={false}
              autoComplete="current-password"
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              id="password"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="password"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        className="MuiBox-root MuiBox-root-86"
      >
        <div
          className="MuiBox-root MuiBox-root-87 makeStyles-btnWrapper-1"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="submit"
          >
            <span
              className="MuiButton-label"
            >
              <span>
                Sign In
              </span>
            </span>
          </button>
        </div>
      </div>
      <div
        className="MuiBox-root MuiBox-root-108"
      >
        <div
          className="MuiGrid-root MuiGrid-container"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
          >
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
              href="#"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
            >
              Forgot password?
            </a>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item"
          >
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
              href="#"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
            >
              No account? Sign Up
            </a>
          </div>
        </div>
      </div>
    </div>
  </form>
</main>
`;
